controllers/routing/routing_controller_test.go

```diff
@@ -343,42 +297,124 @@ var _ = Describe("Platform routing setup for the component", test.EnvTest(), fun
 
 	})
 
-	PWhen("component is deleted all routing resources should be removed", func() {
+	When("component is deleted all routing resources should be removed", func() {
 
+		It("should remove the routing resources when both public;external", func(ctx context.Context) {
+			// given
+			component, errCreate := test.CreateResource(ctx, envTest.Client,
+				componentResource("public-and-external-test-deletion", appNs.Name))
+			Expect(errCreate).ToNot(HaveOccurred())
+			toRemove = append(toRemove, component)
+
+			// when
+			By("adding routing requirements on the resource and related svc", func() {
+				// before updating component, ensure finalizers have been set
+				component = ensureFinalizersSet(ctx, component)
+
+				// required annotation for watched custom resource:
+				// 	routing.opendatahub.io/export-mode: "public;external"
+				exportCustomResource(ctx, component, "public;external")
+
+				// required labels for the exported service:
+				// 	routing.opendatahub.io/exported: "true"
+				// 	platform.opendatahub.io/owner-name: test-component
+				// 	platform.opendatahub.io/owner-kind: Component
+				addRoutingRequirementsToSvc(ctx, svc, component)
+			})
+
+			// then
+			externalResourcesShouldExist(ctx, domain, svc)
+			publicResourcesShouldExist(ctx, svc)
+
+			// when
+			By("deleting the component", func() {
+				// Re-fetch the component from the cluster to get the latest version
+				errGetComponent := envTest.Client.Get(ctx, client.ObjectKey{
+					Namespace: component.GetNamespace(),
+					Name:      component.GetName(),
+				}, component)
+				Expect(errGetComponent).ToNot(HaveOccurred())
+
+				errDelete := envTest.Client.Delete(ctx, component)
+				Expect(errDelete).ToNot(HaveOccurred())
+			})
+
+			// then
+			Eventually(routeExistsFor(svc, domain)).
```

Comment:
I think we should check for `NotFound` error instead of generic one. With generic `ShouldNot(Succeed())` we can pass on any other error that occurred in the assertion, hiding potential regressions.

Reply:
[d1909f9](https://github.com/maistra/odh-platform/pull/39/commits/d1909f93404529fc63c06db80cb53cee639fae93)